apps/vessel-history/features/risk-indicator/risk-indicator.slice.ts

```diff
@@ -35,37 +35,49 @@ export const parseMergedVesselsUniqueId = (id: string): FetchIds[] =>
     const [datasetId, vesselId, tmtId] = x.split('|')
     return { datasetId, vesselId, tmtId }
   })
+
+type ThunkParameters = {
+  idData: FetchIds[]
+  indicator: string
+}
 export const fetchIndicatorsByIdThunk = createAsyncThunk(
   'indicators/fetchById',
-  async (idData: FetchIds[], { getState, rejectWithValue }) => {
+  async (params: ThunkParameters, { getState, rejectWithValue }) => {
+    const { idData, indicator } = params
     try {
       const state = getState() as RootState
       const queryParams = selectEventDatasetsConfigQueryParams(state)
       const query = queryParams
         .map((query) => `${query.id}=${encodeURIComponent(query.value)}`)
         .join('&')
-      const indicator = await GFWAPI.fetch<Indicator>(`/prototype/vessels/indicators?${query}`, {
-        method: 'POST',
-        body: idData.map(({ datasetId, vesselId, tmtId: vesselHistoryId }) => ({
-          ...(datasetId !== NOT_AVAILABLE && { datasetId }),
-          ...(vesselId !== NOT_AVAILABLE && { vesselId }),
-          ...(vesselHistoryId !== NOT_AVAILABLE && { vesselHistoryId }),
-        })) as any,
-        version: '',
-      })
-      indicator.id = getMergedVesselsUniqueId(idData)
-      return indicator
+
+      const result = await GFWAPI.fetch<Indicator>(
+        `/prototype/vessels/indicators?includes=${indicator}&${query}`,
+        {
+          method: 'POST',
+          body: idData.map(({ datasetId, vesselId, tmtId: vesselHistoryId }) => ({
+            ...(datasetId !== NOT_AVAILABLE && { datasetId }),
+            ...(vesselId !== NOT_AVAILABLE && { vesselId }),
+            ...(vesselHistoryId !== NOT_AVAILABLE && { vesselHistoryId }),
+          })) as any,
+          version: '',
+        }
+      )
+
+      result.id = getMergedVesselsUniqueId(idData)
+      return result
     } catch (e: any) {
       return rejectWithValue(parseAPIError(e))
     }
   },
   {
-    condition: (idData: FetchIds[], { getState, extra }) => {
+    condition: (params: ThunkParameters, { getState, extra }) => {
+      const { idData, indicator } = params
       const state = getState() as RootState
       const mergedVesselsUniqueId = getMergedVesselsUniqueId(idData)
       const indicators = selectById(state, mergedVesselsUniqueId)
-      const fetchStatus = selectIndicatorsStatus(state)
-      if (indicators !== undefined || fetchStatus === AsyncReducerStatus.LoadingItem) {
+
+      if (indicators !== undefined && Object.keys(indicators).length >= 6) {
```

Comment:
This is not preventing multiple requests on the same indicator from being triggered for the same vessel

Reply:
how did you reproduce that? I am trying everything and I can't reproduce it

Reply:
This `condition` callback is executed for every call to the `fetchIndicatorsByIdThunk`. Before we were requesting all the indicators for a given Id, so to prevent multiple calls we checked that we already had indicators in the redux store or if there was an ongoing request `indicators !== undefined || fetchStatus === AsyncReducerStatus.LoadingItem`

Now that we are retrieving the indicators individually, we should check if that indicator (individually) has data or if that indicator data is being loaded (this will likely require some additional effort tweaking `createAsyncSlice`)